tests/gar.test.mjs

```diff
@@ -1200,10 +1204,160 @@ describe('GatewayRegistry', async () => {
     });
   });
 
-  // save observations
   describe('Save-Observations', () => {
-    it('should save observations', async () => {
-      // Steps: add a gateway, create the first epoch to prescribe it, submit an observation from the gateway, tick to the epoch distribution timestamp, check the rewards were distributed correctly
+    const distributionTimestamp = genesisEpochTimestamp + distributionDelay;
+    const observerAddress = 'observer-address-'.padEnd(43, 'a');
+
+    let gatewayMemory = sharedMemory;
+    before(async () => {
+      // Join a gateway with the observer
+      const gatewayAddress = 'gateway-address-'.padEnd(43, 'a');
+      const { memory: addGatewayMemory } = await joinNetwork({
+        address: gatewayAddress,
+        memory: sharedMemory,
+        timestamp: genesisEpochTimestamp,
+        observerAddress,
+      });
+
+      // Create the first epoch with distributions already set
+      const futureTick = await handle(
+        {
+          Tags: [{ name: 'Action', value: 'Tick' }],
+          Timestamp: distributionTimestamp, // Tick to when we'll accept observations
+        },
+        addGatewayMemory,
+      );
+
+      // Assert distributions are correct
+      const {
+        totalEligibleObserverReward,
+        totalEligibleGatewayReward,
+        totalEligibleRewards,
+        totalEligibleGateways,
+        rewards,
+      } = JSON.parse(futureTick.Messages[0].Data).maybeNewEpoch.distributions;
+      assert.equal(totalEligibleObserverReward, 1_250_000_000);
+      assert.equal(totalEligibleGatewayReward, 11_250_000_000);
+      assert.equal(totalEligibleRewards, 25_000_000_000);
+      assert.equal(totalEligibleGateways, 2);
+      assert.deepEqual(rewards, {
+        eligible: {
+          '2222222222222222222222222222222222222222222': {
+            operatorReward: 12_500_000_000,
+            delegateRewards: [],
+          },
+          'gateway-address-aaaaaaaaaaaaaaaaaaaaaaaaaaa': {
+            operatorReward: 12_500_000_000,
+            delegateRewards: [],
+          },
+        },
+      });
+
+      // Assert prescribed observers
+      const prescribedObservers = JSON.parse(futureTick.Messages[0].Data)
+        .maybeNewEpoch.prescribedObservers;
+      assert.equal(prescribedObservers.length, 2);
+      assert.equal(prescribedObservers[0].observerAddress, STUB_ADDRESS);
```

Comment:
Will these sort stably all the time?

Reply:
good question. possibly not

Reply:
Fixed in the refactor, they were indeed sorted randomly: [0b61e65](https://github.com/ar-io/ar-io-network-process/pull/250/commits/0b61e65340fcd055492b9d96d2a7bbd781198be0)